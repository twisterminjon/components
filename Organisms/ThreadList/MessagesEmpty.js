import React, { Component } from 'react'
import { Icon } from 'semantic-ui-react'

export default class MessagesEmpty extends Component {
  render() {
    return (
      <div className="messagelist-messagesempty">
        <Icon name="comments" size="massive" />
        <h4 className="messagelist-messagesempty-title">You don't have any messages yet.</h4>
        <p className="messagelist-messagesempty-text">
          To send a message, click the message button <Icon name="comment" />
          next to a patient or staff member in the Staff or Patients tab.
        </p>
      </div>
    )
  }
}
