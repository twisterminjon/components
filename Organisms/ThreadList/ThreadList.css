.messagelist-list {
  min-width: var(--message-sidebar-width);
  max-width: var(--message-sidebar-width);
  height: calc(100% - var(--searchbar-height));

  overflow-y: scroll;

  border-right: 1px solid var(--misc__card__border_color);
}

.messagelist-list {
  height: calc(100% - var(--message-banner-height));
  overflow-y: auto;
}

.messagelist-messagesempty {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  color: var(--brandcolor);
  font-weight: 300;
}

.messagelist-messagesempty-title {
  font-size: 24px;
  max-width: 300px;
  text-align: center;
}

.messagelist-messagesempty-text {
  max-width: 400px;
  text-align: center;
  margin-top: 20px;
}

.messagelist-add-button-background {
  display: flex;
  justify-content: center;
  align-items: center;

  height: 18px;
  width: 18px;

  border-radius: 9px;

  background-color: var(--brandcolor);
}

.messagelist-add-button {
  display: flex;
  justify-content: center;
  align-items: center;

  background-color: var(--messagepage_bg);

  height: 60px;
  width: 60px;
}

.messagelist-add-button:hover {
  background-color: var(--button_color_hover);
}

.messagelist-add-button:focus {
  background-color: var(--button_color_hover);
}

.messagelist-add-button:active {
  transform: scale(1.01);
}

.messagelist-loader .ui.inverted.dimmer {
  background-color: var(--messagepage_bg);
}
