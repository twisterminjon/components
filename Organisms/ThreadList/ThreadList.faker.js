export default {
  rooms: [
    {
      id: 'room1',
      unreadMessages: 0,
      messages: [
        {
          id: 'message1',
          roomId: 'room1',
          timestamp: '2018-12-05T10:30:15',
          text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. ',
          seen: true,
          sender: {
            id: '1111',
            displayName: 'me',
            profileImage: '',
          },
          recipient: {
            id: '1',
            displayName: 'Dr. Stephen Strange',
            profileImage: 'https://www.fillmurray.com/60/60',
          },
        },
      ],
    },
    {
      id: 'room2',
      unreadMessages: 0,
      messages: [
        {
          id: 'message23',
          roomId: 'room1',
          timestamp: '2018-12-01T10:30:15',
          text: 'Debitis error dolorum excepturi sed fugit, in architecto',
          seen: true,
          sender: {
            id: '1',
            displayName: 'Nurse Jacky',
            profileImage: 'https://www.fillmurray.com/100/100',
          },
          recipient: {
            id: '1111',
            displayName: 'me',
            profileImage: '',
          },
        },
      ],
    },
    {
      id: 'room3',
      unreadMessages: 0,
      messages: [
        {
          id: 'message234',
          roomId: 'room1',
          timestamp: '2018-10-29T10:30:15',
          text: 'modi hic qui voluptatem incidunt voluptas quibusdam et quo saepe at alias perferendis labore.',
          seen: true,
          sender: {
            id: '1',
            displayName: 'Dr. Douglas Powers',
            profileImage: 'https://www.fillmurray.com/160/160',
          },
          recipient: {
            id: '1111',
            displayName: 'me',
            profileImage: '',
          },
        },
      ],
    },
    {
      id: 'room4',
      unreadMessages: 0,
      messages: [
        {
          id: 'message1',
          roomId: 'room1',
          timestamp: '2018-10-15T10:30:15',
          text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. ',
          seen: true,
          sender: {
            id: '1111',
            displayName: 'me',
            profileImage: '',
          },
          recipient: {
            id: '1',
            displayName: 'Dr. X',
            profileImage: 'https://www.fillmurray.com/60/60',
          },
        },
      ],
    },
    {
      id: 'room5',
      unreadMessages: 1,
      messages: [
        {
          id: 'message23',
          roomId: 'room1',
          timestamp: '2018-01-01T10:30:15',
          text: 'Debitis error dolorum excepturi sed fugit, in architecto',
          seen: false,
          sender: {
            id: '1',
            displayName: 'Dr. Who',
            profileImage: 'https://www.fillmurray.com/100/100',
          },
          recipient: {
            id: '1111',
            displayName: 'me',
            profileImage: '',
          },
        },
      ],
    },
    {
      id: 'room6',
      unreadMessages: 0,
      messages: [
        {
          id: 'message234',
          roomId: 'room1',
          timestamp: '2018-01-01T10:30:15',
          text: 'modi hic qui voluptatem incidunt voluptas quibusdam et quo saepe at alias perferendis labore.',
          seen: true,
          sender: {
            id: '1',
            displayName: 'Dr. Bruce Banner',
            profileImage: 'https://www.fillmurray.com/160/160',
          },
          recipient: {
            id: '1111',
            displayName: 'me',
            profileImage: '',
          },
        },
      ],
    },
  ],
}
