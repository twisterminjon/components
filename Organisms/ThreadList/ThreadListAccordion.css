.messagelistaccordion {
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: nowrap;

  overflow: hidden;
  height: 60px;
  width: 100%;

  padding-left: 16px;

  background-color: var(--messagepage_bg);

  border-bottom: solid 1px var(--brandcolor);
  text-decoration: none;
  cursor: default; /* don't show a text cursor */
  transition: background-color 250ms ease-in-out, transform 150ms ease;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}

.messagelistaccordion span {
  flex: 1;

  color: var(--button_color_hover);

  text-align: left;
  font-family: var(--font-message-page);
  font-size: var(--font-base-body--size);
  font-weight: 500;

  padding: 8px 12px;

  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.messagelistaccordion-button {
  display: flex;
  justify-content: center;
  align-items: center;

  background-color: var(--messagepage_bg);

  height: 60px;
  width: 60px;
}

.messagelistaccordion-button:hover {
  background-color: var(--button_color_hover);
}

.messagelistaccordion-button:focus {
  background-color: var(--button_color_hover);
}

.messagelistaccordion-button:active {
  transform: scale(1.01);
}

.messagelistaccordion-content {
  overflow: hidden;
  padding-bottom: 20px;
}

.messagelistaccordion--hide {
  height: 0;
  padding-bottom: 0;
}

.messagelistaccordion .button-custom {
  font-size: 16px !important;
}
