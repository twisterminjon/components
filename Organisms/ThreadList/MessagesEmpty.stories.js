import React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs'

import MessagesEmpty from './MessagesEmpty'

const stories = storiesOf('Provider/v1/Organisms/ThreadList', module)
stories.addDecorator(withKnobs)

stories.add('MessagesEmpty', () => {
  return (
    <div
      style={{
        position: 'fixed',
        top: 100,
        right: 0,
        bottom: 0,
        left: 0,
      }}>
      <MessagesEmpty />
    </div>
  )
})
