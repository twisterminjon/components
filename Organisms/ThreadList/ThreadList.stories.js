import React from 'react'
import StoryRouter from 'storybook-react-router'

import { storiesOf } from '@storybook/react'
import { withKnobs, text, select, boolean } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import ThreadList from './ThreadList'
import threads from '../../../Mocks/ThreadList.mock'

const stories = storiesOf('Provider/v1/Organisms/ThreadList', module)
stories.addDecorator(withKnobs)

stories.addDecorator(StoryRouter()).add('ThreadList', () => {
  const selectedIdOpts = {
    room1: 'thread 1',
    room2: 'thread 2',
    room3: 'thread 3',
  }
  const selectedIdVal = select('selectedId', selectedIdOpts, 'thread 2')
  const loadingVal = boolean('loading', false)

  const filterVal = text('filter', '')

  return (
    <div>
      <ThreadList
        threads={threads}
        selectedId={selectedIdVal}
        filter={filterVal}
        onClick={action('onClick')}
        onThreadClick={action('onThreadChange')}
        onSearch={action('onSearch')}
        onShowGroupEditor={action('onShowGroupEditor')}
        hasMoreThreads={false}
        loading={loadingVal}
      />
    </div>
  )
})
