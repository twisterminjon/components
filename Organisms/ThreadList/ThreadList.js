import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { ThreadStatus } from '@shared/helpers'

import ThreadListCard from './ThreadListCard'
import SearchBar from '../../Molecules/SearchBar/SearchBar'
import IconPlus from '../../Atoms/Icons/IconPlus'
import ThreadListAccordion from './ThreadListAccordion'
import Loader from '../../Atoms/Loader/Loader'

import './ThreadList.css'

export default class ThreadList extends Component {
  static propTypes = {
    /** Array of data to display */
    threads: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        unreadMessagesCount: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        isGroup: PropTypes.bool.isRequired,
        isNamedGroup: PropTypes.bool.isRequired,
        groupName: PropTypes.string,
        profileImage: PropTypes.string.isRequired,
        isOwn: PropTypes.bool.isRequired,
        message: PropTypes.string.isRequired,
        timestamp: PropTypes.string.isRequired,
        unread: PropTypes.bool.isRequired,
        membersCount: PropTypes.number,
        status: PropTypes.oneOf(Object.values(ThreadStatus)).isRequired,
        showDraftPreview: PropTypes.bool.isRequired,
        showDraftNotification: PropTypes.bool.isRequired,
      }).isRequired
    ).isRequired,

    /** The currently selected item in the list */
    selectedId: PropTypes.string,

    /** Called after the add group action */
    onShowGroupEditor: PropTypes.func.isRequired,

    /** Function use to filter the threads */
    onSearch: PropTypes.func.isRequired,

    /** Called after thread from the list pressed */
    onThreadClick: PropTypes.func.isRequired,

    /** Indicates if there are more threads to load */
    hasMoreThreads: PropTypes.bool.isRequired,

    /** Indicates if the query is loading */
    loading: PropTypes.bool.isRequired,

    /** Called after a scroll has reached the bottom of the displayed list */
    onReachBottom: PropTypes.func.isRequired,
  }
  static defaultProps = {
    selectedId: '',
  }

  constructor(props) {
    super(props)
    this.state = {
      searchTerm: '',
    }

    this.handleSearchChange = this.handleSearchChange.bind(this)
  }

  handleSearchChange(searchTerm) {
    this.setState({ searchTerm })
    this.props.onSearch(searchTerm)
  }

  render() {
    const { selectedId, onShowGroupEditor, threads, hasMoreThreads, loading, onThreadClick } = this.props

    const { searchTerm } = this.state

    const handleScroll = (entries, observer) => {
      if (entries[0].isIntersecting && hasMoreThreads) {
        this.props.onReachBottom()
      }
    }

    const startObserving = observer => {
      let scrollTarget = document.querySelector('[data-testid="message-list-messages-content"]')
      if (scrollTarget && scrollTarget.lastChild) {
        observer.observe(scrollTarget.lastChild)
      }
    }
    let observer = new IntersectionObserver(handleScroll)
    // If there are more threads start the observer for the last message on the list
    // if not, disconnect the observer and stop fetching more data
    if (hasMoreThreads) startObserving(observer)
    else observer.disconnect()

    const renderThreads = threads.map(thread => {
      return (
        <ThreadListCard
          thread={thread}
          key={thread.id}
          selected={thread.id === selectedId}
          onClick={() => onThreadClick(thread.id)}
        />
      )
    })

    return (
      <div>
        <SearchBar onChange={this.handleSearchChange} value={searchTerm} focusOnMount={false} />

        <div className="messagelist-list" data-testid="message-list">
          <ThreadListAccordion startOpen onClickGroup={onShowGroupEditor} name="messages" label="Messages">
            {renderThreads}
          </ThreadListAccordion>
          <div className="messagelist-loader">
            <Loader show={loading} />
          </div>
        </div>
      </div>
    )
  }
}

export const AddButton = ({ onClick }) => {
  return (
    <button className="messagelist-add-button button-reset" onClick={onClick}>
      <div className="messagelist-add-button-background">
        <IconPlus size={9} color="var(--messagepage_bg)" />
      </div>
    </button>
  )
}
