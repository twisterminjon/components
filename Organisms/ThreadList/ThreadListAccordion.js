import React from 'react'
import PropTypes from 'prop-types'

import './ThreadListAccordion.css'
import Button from '../../Atoms/Button/Button'

ThreadListAccordion.propTypes = {
  /** A name for the control, will be included with on onActivate property */
  name: PropTypes.string.isRequired,

  /** The label for the accordion */
  label: PropTypes.string.isRequired,

  /** Function to create a new Group */
  onClickGroup: PropTypes.func.isRequired,

  /** Primary content. */
  children: PropTypes.node,

  /** ClassName for the wrapper */
  className: PropTypes.string,

  /** Style for the wrapper */
  style: PropTypes.object,
}
ThreadListAccordion.defaultProps = {
  renderControls: null,
  className: '',
  style: {},
}

export default function ThreadListAccordion({ name, label, children, onClickGroup, className = '', style }) {
  return (
    <div className="messagelistaccordion-wrap">
      <div
        className={`messagelistaccordion ${className}`.trim()}
        style={style}
        data-testid={`message-list-accordion-${name}`}>
        <span>{label}</span>
        <Button
          data-testid="message-list-add-group"
          size="medium"
          style={{ fontSize: '12px !important' }}
          onClick={onClickGroup}>
          Add Group
        </Button>
      </div>
      {children && (
        <div className={`messagelistaccordion-content`.trim()} data-testid={`message-list-${name}-content`}>
          {children}
        </div>
      )}
    </div>
  )
}
