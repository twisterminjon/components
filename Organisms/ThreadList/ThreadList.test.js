import React from 'react'
import renderer from 'react-test-renderer'

import ThreadList from './ThreadList'
import threads from '../../../Mocks/ThreadList.mock'

describe('ThreadList', () => {
  const mockFun = jest.fn()

  beforeAll(() => {
    const observe = jest.fn()
    const disconnect = jest.fn()

    window.IntersectionObserver = jest.fn(() => ({
      observe,
      disconnect,
    }))
  })

  it('matches the snapshot', () => {
    // We need to send a consistent timestamp to the component as
    // it will convert the timestamp into a friendly string
    // we want the same string every time the test runs so the snapshot matches
    const now = new Date()
    now.setHours(now.getHours() - 1)
    const timestamp = now.toISOString()

    threads.forEach(thread => (thread.timestamp = timestamp))

    const comp = renderer.create(
      <ThreadList
        threads={threads}
        selectedId="room1"
        onClick={mockFun}
        onShowGroupEditor={mockFun}
        onSearch={mockFun}
        hasMoreThreads={false}
        loading={false}
        onThreadClick={mockFun}
        onReachBottom={mockFun}
      />
    )
    const tree = comp.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
