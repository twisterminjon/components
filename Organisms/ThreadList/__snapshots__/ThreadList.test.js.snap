// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadList matches the snapshot 1`] = `
<div>
  <div
    className="searchbar-wrap"
    data-testid="search-bar"
    style={Object {}}
  >
    <div
      className="searchbar-search"
    >
      <div
        className="searchbar-icon-wrap"
      >
        <svg
          fillRule="evenodd"
          height="14"
          style={Object {}}
          viewBox="0 0 20 20"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="var(--searchbar__icon_fg)"
          >
            <path
              d="M19.7280055,17.2912801 L15.8331894,13.3971292 C15.6573952,13.2213651 15.4190966,13.1237184 15.1690783,13.1237184 L14.532313,13.1237184 C15.6105168,11.7449468 16.2511887,10.0107411 16.2511887,8.12420662 C16.2511887,3.63636364 12.6142039,0 8.12559434,0 C3.63698477,0 0,3.63636364 0,8.12420662 C0,12.6120496 3.63698477,16.2484132 8.12559434,16.2484132 C10.0124511,16.2484132 11.746953,15.6078508 13.1259601,14.5298311 L13.1259601,15.1664876 C13.1259601,15.4164632 13.2236235,15.6547212 13.3994176,15.8304853 L17.2942337,19.7246363 C17.6614481,20.0917879 18.2552415,20.0917879 18.6185493,19.7246363 L19.724099,18.6192755 C20.0913133,18.2521238 20.0913133,17.6584318 19.7280055,17.2912801 Z M8.12559434,13.1237184 C5.36367357,13.1237184 3.12522859,10.8895616 3.12522859,8.12420662 C3.12522859,5.36275754 5.35976704,3.12469485 8.12559434,3.12469485 C10.8875151,3.12469485 13.1259601,5.35885167 13.1259601,8.12420662 C13.1259601,10.8856557 10.8914216,13.1237184 8.12559434,13.1237184 Z"
            />
             
          </g>
        </svg>
      </div>
      <input
        className="searchbar-input"
        defaultValue=""
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
    </div>
  </div>
  <div
    className="messagelist-list"
    data-testid="message-list"
  >
    <div
      className="messagelistaccordion-wrap"
    >
      <div
        className="messagelistaccordion"
        data-testid="message-list-accordion-messages"
        style={Object {}}
      >
        <span>
          Messages
        </span>
        <button
          className="ui medium button button-custom"
          data-testid="message-list-add-group"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          style={
            Object {
              "backgroundColor": "var(--button_color)",
              "color": "var(--dark_black)",
              "fontSize": "12px !important",
            }
          }
          type="submit"
        >
          Add Group
        </button>
      </div>
      <div
        className="messagelistaccordion-content"
        data-testid="message-list-messages-content"
      >
        <button
          className="messagelistcard-wrap"
          data-testid="message-card-Christina Robinson, Paul Jason, RN"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <div
            className="messagelistcard"
          >
            <div
              className="messagelistcard-left"
            >
              <div
                className="messageavatar"
              >
                <svg
                  className=""
                  height="18"
                  style={Object {}}
                  viewBox="0 0 25 18"
                  width="25"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.5 8.854c2.418 0 4.375-1.98 4.375-4.427C11.875 1.98 9.918 0 7.5 0S3.125 1.98 3.125 4.427c0 2.447 1.957 4.427 4.375 4.427zm3 1.265h-.324c-.813.395-1.715.632-2.676.632-.96 0-1.86-.237-2.676-.632H4.5c-2.484 0-4.5 2.04-4.5 4.554v1.138c0 1.047.84 1.897 1.875 1.897h11.25c1.035 0 1.875-.85 1.875-1.897v-1.138c0-2.514-2.016-4.554-4.5-4.554zm8.25-1.265c2.07 0 3.75-1.7 3.75-3.794 0-2.095-1.68-3.795-3.75-3.795-2.07 0-3.75 1.7-3.75 3.795 0 2.094 1.68 3.794 3.75 3.794zm1.875 1.265h-.148c-.543.19-1.118.316-1.727.316-.61 0-1.184-.126-1.727-.316h-.148c-.797 0-1.531.233-2.176.609.953 1.04 1.551 2.419 1.551 3.945v1.517c0 .087-.02.17-.023.253h6.898c1.035 0 1.875-.85 1.875-1.897 0-2.447-1.957-4.427-4.375-4.427z"
                    fill="var(--brandcolor_light)"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
              <span
                className="messagelistcard-status messagelistcard-status--read"
              >
                read
              </span>
            </div>
            <div
              className="messagelistcard-info"
            >
              <p
                className="messagelistcard-label messagelistcard--unread"
                data-testid="message-card-name-Christina Robinson, Paul Jason, RN"
              >
                Christina Robinson, Paul Jason, RN
              </p>
              <div
                className="messagelistcard-message"
              >
                <div
                  className="messagelistcard-draft-bullet"
                />
                <p
                  className="messagelistcard-snippet messagelistcard--unread"
                  data-testid="message-card-snippet-Christina Robinson, Paul Jason, RN"
                >
                  Yo ðŸ¦„
                </p>
              </div>
            </div>
          </div>
          <span
            className="timestamp  messagelistcard-time messagelistcard--unread"
            style={Object {}}
          >
            an hour ago
          </span>
        </button>
        <button
          className="messagelistcard-wrap"
          data-testid="message-card-Jennifer Reyes"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <div
            className="messagelistcard"
          >
            <div
              className="messagelistcard-left"
            >
              <div
                className="avatar"
                style={
                  Object {
                    "height": "32px",
                    "lineHeight": "32px",
                    "width": "32px",
                  }
                }
              >
                <span>
                  J
                </span>
              </div>
              <span
                className="messagelistcard-status"
              >
                sent
              </span>
            </div>
            <div
              className="messagelistcard-info"
            >
              <p
                className="messagelistcard-label messagelistcard--unread"
                data-testid="message-card-name-Jennifer Reyes"
              >
                Jennifer Reyes
              </p>
              <div
                className="messagelistcard-message"
              >
                <span
                  className="messagelistcard-draft-text"
                >
                  Draft:
                </span>
                <p
                  className="messagelistcard-snippet messagelistcard--unread"
                  data-testid="message-card-snippet-Jennifer Reyes"
                >
                  hi jen
                </p>
              </div>
            </div>
          </div>
          <span
            className="timestamp  messagelistcard-time messagelistcard--unread"
            style={Object {}}
          >
            an hour ago
          </span>
        </button>
        <button
          className="messagelistcard-wrap"
          data-testid="message-card-Agnes Rogers, MD"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <div
            className="messagelistcard"
          >
            <div
              className="messagelistcard-left"
            >
              <div
                className="avatar"
                style={
                  Object {
                    "height": "32px",
                    "lineHeight": "32px",
                    "width": "32px",
                  }
                }
              >
                <img
                  alt="user avatar"
                  src="https://www.fillmurray.com/200/200"
                  style={
                    Object {
                      "height": "32px",
                      "lineHeight": "32px",
                      "width": "32px",
                    }
                  }
                />
              </div>
              <span
                className="messagelistcard-status"
              >
                sent
              </span>
            </div>
            <div
              className="messagelistcard-info"
            >
              <p
                className="messagelistcard-label "
                data-testid="message-card-name-Agnes Rogers, MD"
              >
                Agnes Rogers, MD
              </p>
              <div
                className="messagelistcard-message"
              >
                <p
                  className="messagelistcard-snippet "
                  data-testid="message-card-snippet-Agnes Rogers, MD"
                >
                  Howdy
                </p>
              </div>
            </div>
          </div>
          <span
            className="timestamp  messagelistcard-time "
            style={Object {}}
          >
            an hour ago
          </span>
        </button>
        <button
          className="messagelistcard-wrap"
          data-testid="message-card-Bill, Ted"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <div
            className="messagelistcard"
          >
            <div
              className="messagelistcard-left"
            >
              <div
                className="messageavatar"
              >
                <svg
                  className=""
                  height="18"
                  style={Object {}}
                  viewBox="0 0 25 18"
                  width="25"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.5 8.854c2.418 0 4.375-1.98 4.375-4.427C11.875 1.98 9.918 0 7.5 0S3.125 1.98 3.125 4.427c0 2.447 1.957 4.427 4.375 4.427zm3 1.265h-.324c-.813.395-1.715.632-2.676.632-.96 0-1.86-.237-2.676-.632H4.5c-2.484 0-4.5 2.04-4.5 4.554v1.138c0 1.047.84 1.897 1.875 1.897h11.25c1.035 0 1.875-.85 1.875-1.897v-1.138c0-2.514-2.016-4.554-4.5-4.554zm8.25-1.265c2.07 0 3.75-1.7 3.75-3.794 0-2.095-1.68-3.795-3.75-3.795-2.07 0-3.75 1.7-3.75 3.795 0 2.094 1.68 3.794 3.75 3.794zm1.875 1.265h-.148c-.543.19-1.118.316-1.727.316-.61 0-1.184-.126-1.727-.316h-.148c-.797 0-1.531.233-2.176.609.953 1.04 1.551 2.419 1.551 3.945v1.517c0 .087-.02.17-.023.253h6.898c1.035 0 1.875-.85 1.875-1.897 0-2.447-1.957-4.427-4.375-4.427z"
                    fill="var(--brandcolor_light)"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
              <span
                className="messagelistcard-status messagelistcard-status--read"
              >
                read
              </span>
            </div>
            <div
              className="messagelistcard-info"
            >
              <p
                className="messagelistcard-label "
                data-testid="message-card-name-Bill, Ted"
              >
                Bill, Ted
              </p>
              <div
                className="messagelistcard-message"
              >
                <span
                  className="messagelistcard-draft-text"
                >
                  Draft:
                </span>
                <p
                  className="messagelistcard-snippet "
                  data-testid="message-card-snippet-Bill, Ted"
                >
                  Whats Up?
                </p>
              </div>
            </div>
          </div>
          <span
            className="timestamp  messagelistcard-time "
            style={Object {}}
          >
            an hour ago
          </span>
        </button>
        <button
          className="messagelistcard-wrap"
          data-testid="message-card-Batgirl, Batboy"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <div
            className="messagelistcard"
          >
            <div
              className="messagelistcard-left"
            >
              <div
                className="messageavatar"
              >
                <svg
                  className=""
                  height="18"
                  style={Object {}}
                  viewBox="0 0 25 18"
                  width="25"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.5 8.854c2.418 0 4.375-1.98 4.375-4.427C11.875 1.98 9.918 0 7.5 0S3.125 1.98 3.125 4.427c0 2.447 1.957 4.427 4.375 4.427zm3 1.265h-.324c-.813.395-1.715.632-2.676.632-.96 0-1.86-.237-2.676-.632H4.5c-2.484 0-4.5 2.04-4.5 4.554v1.138c0 1.047.84 1.897 1.875 1.897h11.25c1.035 0 1.875-.85 1.875-1.897v-1.138c0-2.514-2.016-4.554-4.5-4.554zm8.25-1.265c2.07 0 3.75-1.7 3.75-3.794 0-2.095-1.68-3.795-3.75-3.795-2.07 0-3.75 1.7-3.75 3.795 0 2.094 1.68 3.794 3.75 3.794zm1.875 1.265h-.148c-.543.19-1.118.316-1.727.316-.61 0-1.184-.126-1.727-.316h-.148c-.797 0-1.531.233-2.176.609.953 1.04 1.551 2.419 1.551 3.945v1.517c0 .087-.02.17-.023.253h6.898c1.035 0 1.875-.85 1.875-1.897 0-2.447-1.957-4.427-4.375-4.427z"
                    fill="var(--brandcolor_light)"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
              <span
                className="messagelistcard-status"
              >
                sent
              </span>
            </div>
            <div
              className="messagelistcard-info"
            >
              <p
                className="messagelistcard-label "
                data-testid="message-card-name-Batgirl, Batboy"
              >
                Batgirl, Batboy
              </p>
              <div
                className="messagelistcard-message"
              >
                <span
                  className="messagelistcard-draft-text"
                >
                  Draft:
                </span>
                <p
                  className="messagelistcard-snippet "
                  data-testid="message-card-snippet-Batgirl, Batboy"
                >
                  did you see that crazy thing ðŸ¦„
                </p>
              </div>
            </div>
          </div>
          <span
            className="timestamp  messagelistcard-time "
            style={Object {}}
          >
            an hour ago
          </span>
        </button>
        <button
          className="messagelistcard-wrap"
          data-testid="message-card-Dr Strange, MD"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <div
            className="messagelistcard"
          >
            <div
              className="messagelistcard-left"
            >
              <div
                className="avatar"
                style={
                  Object {
                    "height": "32px",
                    "lineHeight": "32px",
                    "width": "32px",
                  }
                }
              >
                <img
                  alt="user avatar"
                  src="https://www.fillmurray.com/100/100"
                  style={
                    Object {
                      "height": "32px",
                      "lineHeight": "32px",
                      "width": "32px",
                    }
                  }
                />
              </div>
              <span
                className="messagelistcard-status"
              >
                sent
              </span>
            </div>
            <div
              className="messagelistcard-info"
            >
              <p
                className="messagelistcard-label "
                data-testid="message-card-name-Dr Strange, MD"
              >
                Dr Strange, MD
              </p>
              <div
                className="messagelistcard-message"
              >
                <p
                  className="messagelistcard-snippet "
                  data-testid="message-card-snippet-Dr Strange, MD"
                >
                  I'm the best
                </p>
              </div>
            </div>
          </div>
          <span
            className="timestamp  messagelistcard-time "
            style={Object {}}
          >
            an hour ago
          </span>
        </button>
        <button
          className="messagelistcard-wrap"
          data-testid="message-card-The D-List"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <div
            className="messagelistcard"
          >
            <div
              className="messagelistcard-left"
            >
              <div
                className="messageavatar messageavatar--accent"
              >
                <svg
                  className=""
                  height="16"
                  style={Object {}}
                  viewBox="0 0 24 16"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="var(--brandcolor)"
                    fillRule="evenodd"
                    transform="translate(1 1)"
                  >
                    <path
                      d="M3.291 7.094c1.21 0 2.195-.984 2.195-2.195 0-1.21-.984-2.194-2.195-2.194-1.21 0-2.194.984-2.194 2.194s.984 2.195 2.194 2.195zm15.36 0c1.21 0 2.195-.984 2.195-2.195 0-1.21-.984-2.194-2.195-2.194-1.21 0-2.194.984-2.194 2.194s.984 2.195 2.194 2.195zM19.75 8.19h-2.195c-.603 0-1.148.243-1.546.638 1.382.757 2.362 2.125 2.575 3.75h2.263c.607 0 1.097-.49 1.097-1.097v-1.097c0-1.21-.984-2.194-2.194-2.194zm-8.778 0c2.123 0 3.84-1.718 3.84-3.84S13.094.51 10.971.51c-2.122 0-3.84 1.718-3.84 3.84s1.718 3.84 3.84 3.84zm2.634 1.097h-.285c-.713.343-1.505.549-2.349.549-.843 0-1.632-.206-2.348-.549h-.285c-2.18 0-3.95 1.77-3.95 3.95v.987c0 .909.738 1.646 1.646 1.646h9.875c.908 0 1.645-.737 1.645-1.646v-.987c0-2.18-1.769-3.95-3.95-3.95zm-7.67-.46c-.398-.394-.943-.637-1.546-.637H2.194C.984 8.19 0 9.175 0 10.385v1.097c0 .607.49 1.097 1.097 1.097h2.26c.216-1.625 1.196-2.993 2.578-3.75z"
                    />
                  </g>
                </svg>
              </div>
              <span
                className="messagelistcard-status messagelistcard-status--read"
              >
                read
              </span>
            </div>
            <div
              className="messagelistcard-info"
            >
              <p
                className="messagelistcard-label "
                data-testid="message-card-name-The D-List"
              >
                The D-List
              </p>
              <div
                className="messagelistcard-message"
              >
                <span
                  className="messagelistcard-draft-text"
                >
                  Draft:
                </span>
                <p
                  className="messagelistcard-snippet "
                  data-testid="message-card-snippet-The D-List"
                >
                  Go Cal!
                </p>
              </div>
            </div>
          </div>
          <span
            className="timestamp  messagelistcard-time "
            style={Object {}}
          >
            an hour ago
          </span>
        </button>
      </div>
    </div>
    <div
      className="messagelist-loader"
    >
      <div
        className="ui basic segment"
        style={Object {}}
      >
        <div
          className="ui inverted dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="ui indeterminate inverted text loader"
            >
              Loading
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
